### PR TITLE
Adding NewDataFrame, fixing bug

### PR DESCRIPTION
**Pull request by lilyclements**
@africanmathsinitiative/developers this is ready to review. Here:

* Fixed the bug in #3753
* Minor edit to layout for dlgFromLibrary
* Added a ucrSave to dlgFromLibrary

### CODE REVIEW

File: instat/dlgFromLibrary.vb [hunk 1]
```diff
@@ -158,19 +167,19 @@ Public Class dlgFromLibrary
...
-
         If lstCollection.SelectedItems.Count > 0 Then
             strDataName = CheckString(lstCollection.SelectedItems(0).SubItems(0).Text)
-            ucrBase.clsRsyntax.SetCommandString(strDataName)
-            ucrBase.clsRsyntax.SetAssignTo(strDataName, strTempDataframe:=strDataName)
-            clsDataFunction.AddParameter("X", strDataName)
-        End If
-        TestOkEnabled()
+            ucrBase.clsRsyntax.SetCommandString(strDataName)
+            If ucrNewDataFrameName.bUserTyped = False Then
```
Comment (dannyparsons) [inline instat/dlgFromLibrary.vb]: This should just be `Not ucrNewDataFrameName.bUserTyped`

File: instat/dlgFromLibrary.vb [hunk 2]
```diff
@@ -158,19 +167,19 @@ Public Class dlgFromLibrary
     End Sub
 
     Private Sub lstCollection_SelectedIndexChanged(sender As Object, e As EventArgs) Handles lstCollection.SelectedIndexChanged
-        Dim strDataName As String
```
Comment (dannyparsons) [inline instat/dlgFromLibrary.vb]: Since this is only used within this method it should still be defined here and not for the whole dialog.

Comment (lilyclements) [inline instat/dlgFromLibrary.vb]: Thanks. There was one point I was using it elsewhere so I moved it and then forgot to put it back.

File: instat/dlgFromLibrary.vb (hunk 1 above)
Comment (lilyclements) [inline instat/dlgFromLibrary.vb]: Okay changing this now. But, out of interest, is there a difference in how this affects the code?

**Comment by lilyclements** [comment]
@dannyparsons I've made the changes. Although I have one more query. Line 207 is:

`frmMain.clsRLink.RunScript(clsHelp.ToScript, strComment:="dlgOpenFromLibrary Opening help page for" & " " & lstCollection.SelectedItems(0).Text & " " & "dataset", bSeparateThread:=False, bUpdateGrids:=False)`

However since we changed the default comment for the ucrBase comments to not include the "dlg"/etc, should I change this?

**Comment by dannyparsons** [comment]
Yeah good spot, that can be changed. Instead use the header of the dialog which is `Text` I think.

**Comment by lilyclements** [comment]
Done! This is ready now

**Pull request merged**